nio/responses.py

```diff
@@ -314,31 +315,52 @@ class FileResponse(Response):
     """A response representing a successful file content request.
 
     Attributes:
-        body (bytes): The file's content in bytes.
+        body (bytes, os.PathLike): The file's content in bytes, or location on disk if provided.
         content_type (str): The content MIME type of the file,
             e.g. "image/png".
         filename (str, optional): The file's name returned by the server.
     """
 
-    body: bytes = field()
+    body: Union[bytes, os.PathLike] = field()
     content_type: str = field()
     filename: Optional[str] = field()
 
     def __str__(self):
         return f"{len(self.body)} bytes, content type: {self.content_type}, filename: {self.filename}"
 
     @classmethod
-    def from_data(cls, data, content_type, filename=None):
+    def from_data(cls, data: Union[bytes, os.PathLike, dict], content_type, filename=None):
         """Create a FileResponse from file content returned by the server.
 
         Args:
-            data (bytes): The file's content in bytes.
+            data (bytes, os.PathLike): The file's content in bytes.
             content_type (str): The content MIME type of the file,
                 e.g. "image/png".
+            filename (str, optional): The file's name returned by the server.
```

Comment:
This seems to be extra? 🤔

Reply:
no, afaik this is actually used, see https://github.com/EEKIM10/niobot-test/blob/master/modules/ytdl.py#L311
Unless I misunderstand?